addons/website_sale/controllers/main.py

```diff
@@ -441,7 +441,7 @@ def product(self, product, category='', search='', **kwargs):
         sitemap=False,
     )
     def product_document(self, product_template, document_id):
-        product_template.check_access_rights('read')
+        product_template.browse().check_access('read')
```

Comment:
Why is the browse necessary ?
The `check_access` method must be called on an empty recordset ?

Reply:
It depends on the programmer's intention.  The existing code was only checking access rights on the model.  But, as you point out, not checking the record looks weird.  So the code was wrong, and the new version will be correct :wink: